registration/utils.py

```diff
@@ -80,3 +84,42 @@ def send_account_info_email(application: BetaTestApplication) -> None:
         subject=settings.ACCOUNT_INFO_EMAIL_SUBJECT,
         recipient_list=(user.email,)
     )
+
+
+def is_external_domain_dns_configured(domain: str) -> bool:
+    """
+    Checks that there is proper configuration for the external_domain.
+    This will check for both domain as well as for the widcard subdomain
+    """
+    return is_subdomain_dns_configured(domain) and is_dns_configured(domain)
+
+
+def is_subdomain_dns_configured(domain: str) -> bool:
+    """
+    Checks that the provided domain has a proper DNS configuration
+    for wildcard subdomain i.e. *.{domain}.
+    We can use a randomly generated long subdomain to confirm wildcard
+    configuration.
+    """
+    subdomain = '{}.{}'.format(
+        binascii.hexlify(os.urandom(20)).decode(),
+        domain
+    )
+    return is_dns_configured(subdomain)
+
+
+def is_dns_configured(domain: str) -> bool:
+    """
+    Checks that the provided domain has a proper DNS configuration.
+    There should be a CNAME record present with value haproxy.opencraft.hosting
+    """
+    try:
+        answers = dns.resolver.resolve(domain, 'CNAME')
+        cname_records = [str(rdata) for rdata in answers]  # There should only be one
+        is_configured_properly = [
+            record == 'haproxy.opencraft.hosting'
```

Comment:
we should get  `haproxy.opencraft.hosting` from a constant since this can change, we might as well use another load ballencer.